libs/ktem/ktem/reasoning/prompt_optimization/decompose_question.py

```diff
@@ -0,0 +1,95 @@
+from ktem.llms.manager import llms
+from ktem.reasoning.prompt_optimization.rewrite_question import RewriteQuestionPipeline
+from langchain.prompts import ChatPromptTemplate
+from langchain_core.pydantic_v1 import BaseModel, Field
+
+from kotaemon.base import Document, HumanMessage, Node, SystemMessage
+from kotaemon.llms import ChatLLM
+
+DEFAULT_DECOMPOSE_PROMPT = (
+    "Given the following question, perform query decomposition. "
+    "Given a user question, break it down into the most specific sub questions you can "
+    "which will later help you answer the original question"
+    "Each sub question should be about a single concept/fact/idea."
+    "Keep the sub question as concise as possible. "
+    "Give answer in {lang}\n"
+    "Original question: {question}\n"
+    "Sub questions: "
+)
+system_prompt_template = (
+    "You are an expert at converting user complex questions into sub questions."
+    "You have access to a database of insurance rulebooks and financial reports"
+    "for building LLM-powered applications."
+    "Perform query decomposition. "
+    "Given a user question, break it down into the most specific sub questions you can"
+    "which will help you answer the original question. "
+    "Each sub question should be about a single concept/fact/idea."
+    "If there are acronyms or words you are not familiar with, "
+    "do not try to rephrase them."
+)
+
+
+class SubQuery(BaseModel):
+    """Search over a database of insurance rulebooks or financial reports"""
+
+    sub_query: str = Field(
+        ...,
+        description="A very specific query against the database.",
+    )
```

Comment:
Please provide a more suitable description. I believe there's nothing about this that is specific to `insurance rulebooks or financial reports`.

Also, please avoided using pedantic v1 as v2 is the new standard.